scripts/package.ps1

```diff
@@ -42,4 +43,27 @@ Enable-WSManCredSSP -Force -Role Server
 winrm set winrm/config/client/auth '@{Basic="true"}'
 winrm set winrm/config/service/auth '@{Basic="true"}'
 winrm set winrm/config/service '@{AllowUnencrypted="true"}'
-Write-BoxstarterMessage "winrm setup complete"
\ No newline at end of file
+winrm set winrm/config/winrs '@{MaxMemoryPerShellMB="300"}'
+winrm set winrm/config '@{MaxTimeoutms="1800000"}'
+
+Write-BoxstarterMessage "Reusing RDP self-signed certificate for WinRM HTTPS"
+$SourceStoreScope = 'LocalMachine'
+$SourceStorename = 'Remote Desktop'
+$SourceStore = New-Object  -TypeName System.Security.Cryptography.X509Certificates.X509Store  -ArgumentList $SourceStorename, $SourceStoreScope
+$SourceStore.Open([System.Security.Cryptography.X509Certificates.OpenFlags]::ReadOnly)
+$cert = $SourceStore.Certificates | Where-Object  -FilterScript {
+$_.subject -like '*'
+}
+
+$DestStoreScope = 'LocalMachine'
+$DestStoreName = 'My'
+$DestStore = New-Object  -TypeName System.Security.Cryptography.X509Certificates.X509Store  -ArgumentList $DestStoreName, $DestStoreScope
+$DestStore.Open([System.Security.Cryptography.X509Certificates.OpenFlags]::ReadWrite)
+$DestStore.Add($cert)
+
+$SourceStore.Close()
+$DestStore.Close()
+
+winrm create winrm/config/listener?Address=*+Transport=HTTPS  `@`{Hostname=`"($certId)`"`;CertificateThumbprint=`"($cert.Thumbprint)`"`}
+
```

Comment:
It would be better to put this added block BEFORE lines 43-45. The reason is that as soon as those lines succeed, packer is able to successfully connect and will start the shutdown process. There's a good chance it will still succeed since the shutdown process includes a sysprep and so there is still a small amount of time for the remaining code to complete, but its just safer to always have those three lines at the very end.